src/backend/catalog/namespace.c

```diff
@@ -4450,6 +4450,13 @@ check_search_path(char **newval, void **extra, GucSource source)
 	char	   *rawname;
 	List	   *namelist;
 
+	/* quick exit for babelfish when setting search path in fmgr_security_definer */
+	if (sql_dialect == SQL_DIALECT_TSQL && set_local_schema_for_func_hook
```

Comment:
validate if dialect check is correct here in case of PG to TSQL call

Reply:
we set the dialect in fmgr before calling set_cofig.
and we only change search path for tsql user defined functions. so dialect will be tsql.